pkg/roachprod/roachprod.go

```diff
@@ -184,6 +184,13 @@ func newCluster(
 	return c, nil
 }
 
+// preferLoadBalancer determines if the node selector section in the cluster
```

Comment:
I guess this is a similar point, but `prefer` suggests that actually returning a LB IP is optional, but it seems we return an error below if we can't find a LB.

I actually prefer this semantics (error returned if LB requested but doesn't exist) unless there's strong reason to provide a fallback. In that case, maybe `requestedLoadBalancer` might be a better name.